src/model/resources/storage/storage.jl

```diff
@@ -173,7 +174,11 @@ function storage!(EP::Model, inputs::Dict, setup::Dict)
 
 	# Capacity Reserves Margin policy
 	if CapacityReserveMargin > 0
-		@expression(EP, eCapResMarBalanceStor[res=1:inputs["NCapacityReserveMargin"], t=1:T], sum(derating_factor(gen[y], tag=res)  * (EP[:vP][y,t] + EP[:vCAPRES_discharge][y,t] - EP[:vCHARGE][y,t] - EP[:vCAPRES_charge][y,t])  for y in STOR_ALL))
+		if StorageVirtualDischarge > 0
+			@expression(EP, eCapResMarBalanceStor[res=1:inputs["NCapacityReserveMargin"], t=1:T], sum(derating_factor(gen[y], tag=res)  * (EP[:vP][y,t] + EP[:vCAPRES_discharge][y,t] - EP[:vCHARGE][y,t] - EP[:vCAPRES_charge][y,t])  for y in STOR_ALL))
+		else
+			@expression(EP, eCapResMarBalanceStor[res=1:inputs["NCapacityReserveMargin"], t=1:T], sum(derating_factor(gen[y], tag=res)  * (EP[:vP][y,t] - EP[:vCHARGE][y,t])  for y in STOR_ALL))
+		end
```

Comment:
Could consider using `add_similar_to_expression!` here, since the first expression here is a superset of the second.